### PR TITLE
Fixed the dot character problem on property names

### PR DESCRIPTION
**Pull request by serdar-eric**
As you know elasticsearch does not accept "." character in property names. If exception's Data property contains property names which includes "." character, this couses parse exception on Elasticsearch itself (in my case SQLException object contains HelpLink.ProdName, HelpLink.ProdVer, HelpLink.EvtSrc, HelpLink.EvtID, HelpLink.BaseHelpUrl, HelpLink.LinkId)

But ElasticSearchTarget.cs uses ElasticSearch.Net's bulk API and elasticsearch bulk api responses with "200", not "400". Even if elasticsearch throws parse exception. 

In this scenerio ElasticSearchResponse.Success is alwasy true. But there isn't any record inserted on elasticsearch.

Because of that i just added "ReplaceDot" method before bulk operation that checks all property names and replaces "." character with an "_".

### CODE REVIEW

**Comment by serdar-eric** [comment]
And i've changed nunit framework version to work with nunitadapter for running test methods with Visual Studio Test Explorer.

File: src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs [hunk 1]
```diff
@@ -175,5 +175,26 @@ private object FormPayload(IEnumerable<LogEventInfo> logEvents)
 
             return payload;
         }
+
+        private ExpandoObject ReplaceDot(ExpandoObject obj)
```
Comment (markmcdowell) [inline src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs]: Can you move this to an extension out of this class. Will be easier to maintain.

File: src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs [hunk 2]
```diff
@@ -175,5 +175,26 @@ private object FormPayload(IEnumerable<LogEventInfo> logEvents)
...
+        {
+            var clone = new ExpandoObject();
+            foreach (var item in obj)
+            {
+                if (item.Value == null) continue;
+
+                if (item.Value.GetType() == typeof(ExpandoObject))
+                    ((IDictionary<string, object>)clone)[item.Key.Replace('.', '_')] 
+                        = ReplaceDot(item.Value as ExpandoObject);
+                else
```
Comment (markmcdowell) [inline src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs]: Use `else if` and `else` to reduce nesting.

**Comment by serdar-eric** [comment]
Thank you Mark. 
I've created a new pull request including changes with your suggestions.

**Comment by markmcdowell** [comment]
Great thanks, merged. 2.1.1 was published with this in.

**Pull request merged**